subprojects/code-quality/src/main/groovy/org/gradle/api/plugins/quality/CheckstylePlugin.java

```diff
@@ -51,8 +55,7 @@ protected String getToolName() {
     protected CodeQualityExtension createExtension() {
         extension = project.getExtensions().create("checkstyle", CheckstyleExtension.class, project);
         extension.setToolVersion(DEFAULT_CHECKSTYLE_VERSION);
-
-        extension.setConfigDir(project.file("config/checkstyle"));
+        extension.getConfigDirectory().set(determineConfigurationDirectory());
```

Comment:
We should fix the call to the Checkstyle task below to use this property.

```
task.setConfigDir(project.provider(new Callable<File>() {
            @Override
            public File call() {
                return extension.getConfigDir();
            }
        }));
```
becomes
```
task.setConfigDir(extension.getConfigDirectory());
```

We should clean-up the Checkstyle task too (to follow our Provider API recommendations).

Reply:
I think this could easily be done in a separate PR.